lib/mongo/collection/view/writable.rb

```diff
@@ -107,6 +109,7 @@ def find_one_and_update(document, opts = {})
           cmd[:maxTimeMS] = max_time_ms if max_time_ms
           cmd[:bypassDocumentValidation] = !!opts[:bypass_document_validation]
           cmd[:writeConcern] = write_concern.options if write_concern
+          cmd[:arrayFilters] = opts[:array_filters] if opts[:array_filters]
```

Comment:
Don't we need to check if the server supports array filters in the findAndModify methods?